crates/sierra-casm/src/lib.rs

```diff
@@ -19,23 +17,23 @@ pub fn compile(mut sierra_json: Value) -> Result<CasmContractClass, String> {
 
     macro_rules! compile_contract {
         ($sierra_type:ty, $casm_type:ty) => {{
-            if let Ok(sierra_class) = serde_json::from_value::<$sierra_type>(sierra_json.clone()) {
-                if let Ok(casm_class) = <$casm_type>::from_contract_class(sierra_class, true) {
-                    return Ok(old_casm_to_newest_casm::<$casm_type>(&casm_class));
-                }
-            }
+            let sierra_class = serde_json::from_value::<$sierra_type>(sierra_json.clone()).unwrap();
+            let casm_class = <$casm_type>::from_contract_class(sierra_class, true).unwrap();
+            return Ok(old_casm_to_newest_casm::<$casm_type>(&casm_class));
```

Comment:
Couldn't this just be a (generic) function or something like that? I like macros, but I'm unsure if this one provides any benefit.

Reply:
Unfortunately, it provides :/ It wouldn't be possible to use `T::from_contract_class` in the function body, because that function is not in the the trait